### PR TITLE
[Client-gen] Remove the use of namespaceifscoped from typed clients

### PR DESCRIPTION
**Pull request by caesarxuchao**
pkg/client/unversioned/events.go unnecessarily uses NamespaceIfScoped many times, I removed those usage so events.go looks more similar to the generated clients.

I also remove the use of NamespaceIfScoped from the client-gen template.

Note: the Create and Update methods in events.go are still unique, the request is scoped to namespace of the passed in API object rather than the event client's namesapce, so they still have to be manually written.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/19307)

### CODE REVIEW

File: pkg/client/unversioned/events.go [hunk 1]
```diff
@@ -120,7 +120,7 @@ func (e *events) Patch(incompleteEvent *api.Event, data []byte) (*api.Event, err
 func (e *events) List(opts api.ListOptions) (*api.EventList, error) {
 	result := &api.EventList{}
 	err := e.client.Get().
-		NamespaceIfScoped(e.namespace, len(e.namespace) > 0).
+		Namespace(e.namespace).
```
Comment (nikhiljindal) [inline pkg/client/unversioned/events.go]: What about the unnamespaced LIST and WATCH?

File: pkg/client/unversioned/events.go [hunk 2]
```diff
@@ -144,7 +144,7 @@ func (e *events) Get(name string) (*api.Event, error) {
 func (e *events) Watch(opts api.ListOptions) (watch.Interface, error) {
 	return e.client.Get().
 		Prefix("watch").
-		NamespaceIfScoped(e.namespace, len(e.namespace) > 0).
+		Namespace(e.namespace).
```
Comment (nikhiljindal) [inline pkg/client/unversioned/events.go]: Same - What about the unnamespaced LIST and WATCH?

File: pkg/client/unversioned/events.go (hunk 1 above)
Comment (caesarxuchao) [inline pkg/client/unversioned/events.go]: When building the URL, if `len(e.namespace)==0`, the namespace won't be shown in the path. That's how all other clients (e.g., the client for Pod) work.

Comment (caesarxuchao) [inline pkg/client/unversioned/events.go]: https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/request.go#L558

**Comment by nikhiljindal** [comment]
LGTM, thanks!

**Comment by nikhiljindal** [comment]
@k8s-bot unit test this

**Comment by caesarxuchao** [comment]
Adding back the LGTM after rebase.

**Comment by caesarxuchao** [comment]
@k8s-bot e2e test this please

**Comment by gmarek** [comment]
e2e failure: #19299
@k8s-bot test this

**Comment by smarterclayton** [comment]
This PR should remove NamespaceIfScoped.  It was only used in v1beta1 and v1beta2.

**Comment by caesarxuchao** [comment]
@smarterclayton, I think NamespaceIfScoped is still needed in Helper: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/resource/helper.go#L58

**Comment by smarterclayton** [comment]
Ok, nm then.

**Pull request merged**